### PR TITLE
WT-2308 Fix for joins on indices that use extractors and 'u' at end of format

### PR DESCRIPTION
**Pull request by ddanderson**
Passes the submitted test case.

### CODE REVIEW

**Comment by ddanderson** [comment]
@michaelcahill, can you review this please?

**Comment by ddanderson** [comment]
@michaelcahill, please hold off on any review until I can get to the bottom of the latest test case from Denis in WT-2308.

File: src/packing/pack_impl.c [hunk 1]
```diff
@@ -164,7 +164,7 @@ __wt_struct_repack(WT_SESSION_IMPL *session, const char *infmt,
 	 * The former case has the size embedded before the item, the latter
 	 * does not.
 	 */
-	if ((len = strlen(outfmt)) > 1 && outfmt[len - 1] == 'u' &&
+	if ((len = strlen(outfmt)) > 0 && outfmt[len - 1] == 'u' &&
```
Comment (michaelcahill) [inline src/packing/pack_impl.c]: @ddanderson, I know you're still bug hunting, but this part of the change confuses me.  Won't it turn ordinary "u" into "U"?  In other words, the common case of a `WT_ITEM` will need to be encoded with a size prefix.

**Comment by michaelcahill** [comment]
@ddanderson, just doing a review, and the first changes for WT-2308 have already been merged into MongoDB master.  Given that, we'll need a new ticket for the remaining fix to avoid confusion the release management process.

**Comment by ddanderson** [comment]
See #2487, that has a comprehensive fix for all tests.

**Pull request closed without merging**